src/Toolbox/Util/Graph/BFS.hs

```diff
@@ -0,0 +1,134 @@
+{-# LANGUAGE BangPatterns #-}
+
+-- | Breadth-First-Search (BFS)
+module Toolbox.Util.Graph.BFS
```

Comment:
You're not importing this in any (non-test) modules? So this is all dead code?

Reply:
This BFS search is used in the next PR #25